### PR TITLE
Extract genKeyPair from GenKeyPair

### PR DESCRIPTION
**Pull request by mem**
genKeyPair takes inputs and produces either a new key (entity) or an
error. It does not ask questions or handle user interaction.

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

### CODE REVIEW

File: pkg/sypgp/sypgp.go [hunk 1]
```diff
@@ -443,54 +443,66 @@ func GetPassphrase(message string, retries int) (string, error) {
...
+	}
+
+	if err = StorePubKey(entity); err != nil {
+		return nil, err
+	}
+
+	return entity, nil
+}
+
+// GenKeyPair generates an OpenPGP key pair and store them in the sypgp home folder
```
Comment (WestleyK) [inline pkg/sypgp/sypgp.go]: Maybe?...

```suggestion
// GenKeyPair generates an PGP key pair and store them in the sypgp home folder
```

File: pkg/sypgp/sypgp.go [hunk 2]
```diff
@@ -443,54 +443,66 @@ func GetPassphrase(message string, retries int) (string, error) {
...
 	conf := &packet.Config{RSABits: 4096, DefaultHash: crypto.SHA384}
 
-	if err = PathsCheck(); err != nil {
-		return
+	entity, err := openpgp.NewEntity(name, comment, email, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	// encrypt private key
```
Comment (WestleyK) [inline pkg/sypgp/sypgp.go]:
```suggestion
	// Encrypt private key
```

To be consistent.

**Review by jscook2345 — APPROVED** [review]
LGTM

**Review by WestleyK — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**